spec/spec.md

```diff
@@ -516,42 +516,130 @@ The following two serviceEndpoints MUST be supported in the DID Document, but on
+Below is a non-normative example of credentialStatus property defined in VC: 
+```json
+"credentialStatus": {	
+  "id": "Qmdfr32sdf32546...",
+  "type": "StatusList2021",
+ "statusPurpose": "revocation",
+  "statusListIndex": "94567",
+  "statusListCredential": 'did:ion:123?service=IdentityHub&queries=W3sibWV0aG9kIjoiQ29sbGVjdGlvbnNRdWVyeSIsI…..'
+}
+```
 
-When `credentialStatus` is deinfed it MUST use StatusList2021 , as defined in section 5.1 of [[ref: Status List 2021]].
+**Construct the Request Object:**
 
-StatusList2021 MUST be discovered using either DID Relative URLs stored in an ID Hub or HTTPS URL. 
+* Create a JSON object for the request.
+* The Request Object MUST include a id property, and its value MUST be an [RFC4122] UUID Version 4 string to identify the request.
+* The Request Object MUST include a target property, and its value MUST be the Decentralized Identifier base URI of the DID-relative URL.
+* The Request Object MUST include a messages property, and its value MUST be an array composed of Message objects that are generated by parsing the DID-relative URL’s queries parameter value as a JSON array and performing the following steps for each entry.:
+  + Construct a Message object.
+  + Set the descriptor property of the Message object to the entry, ensuring it is a valid Message Descriptor object.
+  + Augment the Message object with any signing and authorization values required, as described in the Messages section.
+  + Append the object to the Request Object’s messages array.
 
-An Issuer of a VC MAY have an ID Hub serviceEndpoint in the Issuer's DID Document. ID Hubs are the single endpoint to look up objects associated with a DID, as defined in [Identity-Hub].
-Below is a non-normative example of a DID Document that includes a serviceEndpoint:
+***HTTP POST example:***
 
 ```json
-"service": [
-      {
-        "id": "hubs",
-        "type": "IdentityHub",
-        "serviceEndpoint": [
-          "https://hubs.microsoft.com",
-          "https://datastore.protonmail.com"
-        ]
+POST https://hub.example.com/
+
+BODY {
+  "requestId": "c5784162-84af-4aab-aff5-f1f8438dfc3d",
+  "target": "did:example:123",
+  "messages": [
+    {
+      "descriptor": {
+        "method": "CollectionsQuery",
+        "objectId": "b6464162-84af-4aab-aff5-f1f8438dfc1e"
       }
-]
+    },
+    {...}
+  ]
+}
 ```
 
+#### HTTPS URL Structure
+An Issuer of a VC may issue VCs with HTTPS URL defined in credentialStatus for statusListCredential. Below is a non-normative example of credentialStatus property defined in VC: 
 ```json
+"credentialStatus":
 {
-  "credentialStatus": {
-    "id": "Qmdfr32sdf32546...",
-    "type": "StatusList2021",
-    "statusListIndex": "94567",
-    "statusListCredential": 'did:ion:123?service=IdentityHub&relativeRef=?messages=[{ type: "CollectionsQuery", statement: { id: "Qmdfr32sdf32546..." }}]'
+    "id": " https://example.com/credentials/status/3#234243",
+    "type": " RevocationList2021Status",
```

Comment:
What is `RevocationList2021Status`? This doesn't appear in the referenced spec: https://w3c-ccg.github.io/vc-status-list-2021/

Reply:
Assume this is supposed to be `StatusList2021Entry`

Reply:
Underlying Status List 2021 spec was changed in this commit: https://github.com/w3c-ccg/vc-status-list-2021/commit/a37a8337109920652a69251f2fcb35939060a57f

From `RevocationList2021Status` to `StatusList2021Entry`

Reply:
this is the commit status list is pinned to: https://github.com/w3c-ccg/vc-status-list-2021/tree/343b8b59cddba4525e1ef355356ae760fc75904e

Reply:
see issue #19

Reply:
reasoning here why `RevocationList2021Status`: https://github.com/decentralized-identity/jwt-vc-presentation-profile/pull/56/files#r997429907